File: ios/FincodeViewModule.m
//
//  FincodeViewModule.m
//  Fincode_Manual_Import
//
//  Created by 中嶋彰 on 2023/02/27.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(FincodeViewModule, NSObject)

// UIコンポーネント利用の際で、決済・カード登録・カード更新をReact(JS)から呼び出せるように定義する (React → swift)
RCT_EXTERN_METHOD(initPayment:(NSString *)viewType
                  authorization:(NSString *)authorization
                  apiKey:(NSString *)apiKey
                  apiVersion:(NSString *)apiVersion
                  customerId:(NSString *)customerId
                  payType:(NSString *)payType
                  accessId:(NSString *)accessId
                  id:(NSString *)id)

RCT_EXTERN_METHOD(initCardRegister:(NSString *)viewType
                  authorization:(NSString *)authorization
                  apiKey:(NSString *)apiKey
                  apiVersion:(NSString *)apiVersion
                  customerId:(NSString *)customerId
                  defaultFlg:(NSString *)defaultFlg)

RCT_EXTERN_METHOD(initCardUpdate:(NSString *)viewType
                  authorization:(NSString *)authorization
                  apiKey:(NSString *)apiKey
                  apiVersion:(NSString *)apiVersion
                  customerId:(NSString *)customerId
                  defaultFlg:(NSString *)defaultFlg
                  cardId:(NSString *)cardId)

@end

File: ios/FincodeViewModule.swift
//
//  FincodeViewModule.swift
//  Fincode_Manual_Import
//
//  Created by 中嶋彰 on 2023/02/27.
//

import Foundation
import FincodeSDK

@objc(FincodeViewModule)
class FincodeViewModule: NSObject { 
   
  /// UIコンポーネント利用の際で、決済向けUIの初期化を行う
  @objc public func initPayment(_ viewType: String, authorization: String, apiKey: String, apiVersion: String, customerId: String, payType: String, accessId: String, id: String) {
    
    let rct: RCTFincodeCommonView? =  FincodeViewHolder.sharedInstance.getView(viewType)
    rct?.initPayment(authorization, apiKey: apiKey, apiVersion: apiVersion, customerId: customerId, payType: payType, accessId: accessId, id: id)
  }
  
  /// UIコンポーネント利用の際で、カード登録向けUIの初期化を行う
  @objc public func initCardRegister(_ viewType: String, authorization: String, apiKey: String, apiVersion: String, customerId: String, defaultFlg: String) {
    
    let rct: RCTFincodeCommonView? =  FincodeViewHolder.sharedInstance.getView(viewType)
    rct?.initCardRegister(authorization, apiKey: apiKey, apiVersion: apiVersion, customerId: customerId, defaultFlg: defaultFlg)
  }
  
  /// UIコンポーネント利用の際で、カード更新向けUIの初期化を行う
  @objc public func initCardUpdate(_ viewType: String, authorization: String, apiKey: String, apiVersion: String, customerId: String, defaultFlg: String, cardId: String) {
    
    let rct: RCTFincodeCommonView? =  FincodeViewHolder.sharedInstance.getView(viewType)
    rct?.initCardUpdate(authorization, apiKey: apiKey, apiVersion: apiVersion, customerId: customerId, defaultFlg: defaultFlg, cardId: cardId)
  }
}
